simulator/src/models/power-plant.ts

```diff
@@ -0,0 +1,564 @@
+        let id = uuid.v4();
+        let startDelay = randomFloat(1000, 10000);
+        let stopDelay = randomFloat(1000, 10000);
+        let productionLevel = randomFloat(100, 200);
+        let productionCapacity = randomFloat(200, 400);
+        let productionVariant = randomFloat(2, 10);
+        let productionRatio = 0.1;
+        let batteryCapacity = randomFloat(2000, 5000);
+        return new PowerPlant(id,
+                        owner,
+                        startDelay, 
+                        stopDelay, 
+                        productionLevel,
+                        productionCapacity,
+                        productionVariant,
+                        productionRatio,
+                        batteryCapacity);
+    }
+
+
+    /**
+     * Tries to find a power plant object in the database with the given id.
+     * @returns {Promise<PowerPlant>} the wind object is found
+     */
+    static async findById(id: number): Promise<PowerPlant> {
+        let rows = await ElectricityGridDB.table('Power_plant').select([], [eq('id', id)]);
+        if (rows.length == 1) {
+            let row = rows[0];
+            let plant = new PowerPlant(row.id,
+                            row.owner,
+                            row.start_delay,
+                            row.stop_delay,
+                            row.production_level,
+                            row.production_capacity,
+                            row.production_variant,
+                            row.production_ratio,
+                            row.battery_capacity,
+                            row.time,
+                            row.created_at,
+                            row.updated_at);
+            return plant
+        } else {
+            return Promise.reject("Could not find any wind object with id " + id);
+        }
+    }
+
+    /**
+     * Store a given power plant data in the power_plant_data table.
+     * If there already exists data for this time then update instead.
+     */
+    store() {
+        ElectricityGridDB.table('power_plant').insert_or_update({
+            id: this._id,
+            owner: this._owner,
+
+            start_delay: this._startDelay,
+            stop_delay: this._stopDelay,
+
+            production_level: this._productionLevel,
+            production_capacity: this._productionCapacity,
+            production_variant: this._productionVariant,
+            production_ratio: this._productionRatio,
+        
+            battery_capacity: this._battery.capacity,
+
+            time: this._time,
+            created_at: this._createdAt,
+            updated_at: this._updatedAt,
+        }, ['id']);
+    }
+
+
+    /**
+     * Starts the power plant.
+     * @note Takes some time before its starts.
+     * @note The power plant can only start if its Status is equal Stopped.
+     */
+    start() {
+        if (this._status == Status.Stopped) {
+            this._status = Status.Starting;
```

Comment:
If you have a getter for a variable you may use that instead, consider changing this._status to this.status some goes for other fields where this makes sense.